### PR TITLE
Allow ROOTFSCFG= to not be at the beginning of the command line.

### PR DESCRIPTION
**Pull request by boardwalk**
syslinux passes the entire command line ala main(), ex:
"/myprogram.bin ROOTFSCFG=/json.cfg"

### CODE REVIEW

File: lib/librumprun_base/rumpconfig.c [hunk 1]
```diff
@@ -613,17 +613,22 @@ getcmdlinefromroot(const char *cfgname)
...
 static const size_t rootcfglen = sizeof(ROOTCFG)-1;
-int
-rumprun_config_isonrootfs_p(char *config)
+char *
+rumprun_config_path(char *cmdline)
 {
+	char *cfg = strstr(cmdline, ROOTCFG);
 
-	return strncmp(config, ROOTCFG, rootcfglen) == 0;
+	if(cfg != NULL)
```
Comment (anttikantee) [inline lib/librumprun_base/rumpconfig.c]: We use `if<space>(` formatting.

**Comment by anttikantee** [comment]
Please add yourself to AUTHORS.  Otherwise ok.

**Comment by anttikantee** [comment]
Thanks!

(btw, I don't get a notification if you update your branch, so for future reference please ping the pull req when you complete an interation)

**Comment by boardwalk** [comment]
Sure thing. Thanks. rumprun works fully with syslinux with this change, by the way.

(syslinux feels a bit more minimal than something generated with grub-mkrescue, which added like 20 MBs to my ISO by default...)

**Comment by anttikantee** [comment]
On 24/09/15 13:36, Dan Skorupski wrote:
> Sure thing. Thanks. rumprun works fully with syslinux with this change, by the way.

Any chance you could post about this development to the rumpkernel-users 
list for the benefit of people who don't know about syslinux (e.g. 
myself not too many minutes ago ;).

> (syslinux feels a bit more minimal than something generated with grub-mkrescue, which added like 20 MBs to my ISO by default...)

Should be offer it as an alternative to grub-mkrescue?  Or ditch 
grub-mkrescue entirely?  What are the tradeoffs?
(also excellent things to cover in a mailing list post ;)

Thanks.

**Comment by boardwalk** [comment]
I can try'n find time to make a mailing list post and answer those questions, sure :). No promises though, I've been busy.

**Comment by anttikantee** [comment]
No worries if you don't have time right this second, the mailing list isn't going anywhere ;-)

**Comment by mato** [comment]
@boardwalk You say that rumprun kernels now work with syslinux -- I can't see anything in syslinux documentation about multiboot support. Am I missing something?

**Comment by boardwalk** [comment]
http://www.syslinux.org/wiki/index.php/Mboot.c32

My configuration looks like this:
```
% cat syslinux.cfg
DEFAULT mboot.c32 /tenchi.bin ROOTFSCFG=/config.json
```

**Pull request merged**